golang/vaas/cmd/git-scan/main.go

```diff
@@ -30,14 +30,23 @@ func main() {
 	}
 	log.Println("targetBranch:", targetBranch)
 
-	clientID, exists := os.LookupEnv("VAAS_CLIENT_ID")
-	if !exists {
-		log.Fatal("no Client ID set")
+	clientID, clientIdExists := os.LookupEnv("VAAS_CLIENT_ID")
+	username, usernameExists := os.LookupEnv("VAAS_USERAME")
+	password, passwordExsits := os.LookupEnv("VAAS_PASSWORD")
+	clientSecret, clientSecretExists := os.LookupEnv("VAAS_CLIENT_SECRET")
+
+	if !clientIdExists {
+		log.Fatal("no client_id set")
+
 	}
-	clientSecret, exists := os.LookupEnv("VAAS_CLIENT_SECRET")
-	if !exists {
-		log.Fatal("no Client Secret set")
+	if clientIdExists && (!passwordExsits && !clientSecretExists) {
+		log.Fatal("either password or client_secret must be set")
 	}
+
+	if usernameExists && (!passwordExsits || !clientIdExists) {
+		log.Fatal("when using the username, the password and client_id must be set")
+	}
```

Comment:
We log fatal errors here, but we don't actually abort, right?

Reply:
Fatal does os.exit

https://pkg.go.dev/log#Fatal